File: sketches/sketches.model/src/main/java/edu/kit/kastel/lissa/sketches/model/elements/uml/IUMLInterface.java
package edu.kit.kastel.lissa.sketches.model.elements.uml;

import edu.kit.kastel.lissa.sketches.model.elements.IBox;
import edu.kit.kastel.lissa.sketches.model.impl.elements.uml.UMLInterface;
import edu.kit.kastel.lissa.sketches.model.types.SketchBoxTypes;
import edu.kit.kastel.lissa.sketches.model.types.SketchBoxTypeMapping;

@SketchBoxTypeMapping(type = SketchBoxTypes.INTERFACE, implementation = UMLInterface.class)
public interface IUMLInterface extends IUMLThing, IBox {

}

File: sketches/model/src/main/kotlin/edu/kit/kastel/lissa/sketches/model/elements/uml/IUMLInterface.kt
package edu.kit.kastel.lissa.sketches.model.elements.uml

import edu.kit.kastel.lissa.sketches.model.elements.IBox
import edu.kit.kastel.lissa.sketches.model.impl.elements.uml.UMLInterface
import edu.kit.kastel.lissa.sketches.model.types.SketchBoxTypeMapping
import edu.kit.kastel.lissa.sketches.model.types.SketchBoxTypes

@SketchBoxTypeMapping(type = SketchBoxTypes.INTERFACE, implementation = UMLInterface::class)
interface IUMLInterface : IUMLThing, IBox
